### PR TITLE
rgw_sal_motr: [CORTX-30362] Add & Update User Stats Index Table

### PR DESCRIPTION
**Pull request by sumedhak27**
Changes:
1. Add/Delete User Stats index table on User creation/deletion. [commit 1](https://github.com/Seagate/cortx-rgw/pull/215/commits/4b46b9159e57ec5c6cc0c6c04ca3396dacb0ff6a)
2. Add/Remove bucket stats entry in the index table on bucket creation/deletion. [commit 1](https://github.com/Seagate/cortx-rgw/pull/215/commits/4b46b9159e57ec5c6cc0c6c04ca3396dacb0ff6a)
3. Implement read bucket stats. [commit 2](https://github.com/Seagate/cortx-rgw/pull/215/commits/ce8ecc952604dba4bb18251ab57795bc4d284dcb)
4. Update the total size and object count on object put and delete operations. [commit 3](https://github.com/Seagate/cortx-rgw/pull/215/commits/29ea0b1f915dedd1d35b5b382564d483cf82f8f3)

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/rgw/rgw_sal_motr.cc [hunk 1]
```diff
@@ -2727,15 +2839,49 @@ int MotrAtomicWriter::complete(size_t accounted_size, const std::string& etag,
...
+  if (rc != 0) {
+    ldpp_dout(dpp, 20) << __func__ << ": Failed to get the bucket header."
+      << " bucket = " << tenant_bkt_name << ", ret = " << rc << dendl;
+    return rc;
+  }
+  
+  bufferlist::const_iterator bitr = bl.begin();
+  bkt_header.decode(bitr);
+  rgw_bucket_category_stats& bkt_stat = bkt_header.stats[RGWObjCategory::Main];
+  bkt_stat.total_size += total_data_size - old_obj.get_obj_size();
```
Comment (sachinpunadikar) [inline src/rgw/rgw_sal_motr.cc:2865]: This should be in if block  .. if not versioning_enabled

Comment (sumedhak27) [reply src/rgw/rgw_sal_motr.cc:2865]: It does not matter, if the old object does not exist, its size is defaulted to 0.

Comment (sumedhak27) [reply src/rgw/rgw_sal_motr.cc:2865]: got the mistake 👍🏼

Comment (sumedhak27) [reply src/rgw/rgw_sal_motr.cc]: addressed @sachinpunadikar, tested with the versioned` object as well.
```
╰─➤  aws s3api list-object-versions --bucket mybkt-v1 --profile sumedh
{
    "Versions": [
        {
            "ETag": "\"da39978011e9e121b96e74bc4cf0c737\"",
            "Size": 142044,
            "StorageClass": "STANDARD",
            "Key": "obj1",
            "VersionId": "zWszE5bg845PS9aMMI3Py2R3TBr0mYs",
            "IsLatest": true,
            "LastModified": "2022-05-27T05:59:39.977Z",
            "Owner": {
                "DisplayName": "SumedhAK",
                "ID": "sumedh"
            }
        },
        {
            "ETag": "\"fb92234566bc9f01b853318711106da7\"",
            "Size": 1024,
            "StorageClass": "STANDARD",
            "Key": "obj1",
            "VersionId": "zWszEwvEVllzAOCv9UaipHoEK0Hjxgu",
            "IsLatest": false,
            "LastModified": "2022-05-27T05:56:15.036Z",
            "Owner": {
                "DisplayName": "SumedhAK",
                "ID": "sumedh"
            }
        },
        {
            "ETag": "\"9cce32b2829b5c73b70075cef7210458\"",
            "Size": 1048576,
            "StorageClass": "STANDARD",
            "Key": "obj1",
            "VersionId": "zWszF3WPjn5lcYzjq4dMnj7G6e6vzuk",
            "IsLatest": false,
            "LastModified": "2022-05-27T05:55:49.679Z",
            "Owner": {
                "DisplayName": "SumedhAK",
                "ID": "sumedh"
            }
        }
    ]
}
```

```
"usage": {
        "rgw.main": {
            "size": 1191644,
            "size_actual": 1191936,
            "size_utilized": 1191644,
            "size_kb": 1164,
            "size_kb_actual": 1164,
            "size_kb_utilized": 1164,
            "num_objects": 3
        }
    }
```

Comment (jjxsg) [reply src/rgw/rgw_sal_motr.cc]: This looks alright.

**Review by sachinpunadikar — APPROVED** [review]
+1

**Review by jjxsg — APPROVED** [review]
As discussed, please add a TODO for handling stats in case of deletion of versioned object. 
@sachinpunadikar we may need a story to handle this for versioning team as their implementation of deletion of versioned object is still in progress.

**Pull request merged**